runtime/runtime.go

```diff
@@ -223,6 +225,55 @@ func (r *Runtime) UpdateInstanceConnector(ctx context.Context, instanceID, name
 	return r.EditInstance(ctx, inst, false)
 }
 
+func (r *Runtime) ReloadConfig(ctx context.Context, instanceID string, disableRestartCtrl bool) error {
```

Comment:
It would be nice if we could make this safe for relatively frequent calls – that would enable a) running periodic config syncs in the background for added resilience, b) safety when two calls/mutations happen quickly after each other, such as adding multiple new variables with separate CLI calls.

I think there's a couple things necessary for that:
1. Use a lock or singleflight or something like that to avoid concurrent calls. That would prevent race conditions when e.g. variables are updated rapidly after each other. 
2. Maybe avoid the forced repo refresh on each call? Or is it cheap now? One option might be to propagate the `UpdatedOn` from the admin's deployment and using that as a kind of sequence number to check if we have already stored the latest config or not.